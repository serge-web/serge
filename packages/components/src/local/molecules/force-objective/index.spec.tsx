/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import ForceObjective from './index'

it('ForceObjective renders correctly', () => {
  const tree = renderer
    .create(
      <ForceObjective
        force={{
          color: '#3dd0ff',
          name: 'Blue',
          iconURL: 'http://localhost:8080/default_img/forceDefault.png',
          overview: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est, praesentium sed! Animi atque incidunt maiores nisi vel voluptate? Adipisci architecto aspernatur earum error incidunt laborum magni neque odio repellendus, voluptate!'
        }}
        selectedRoleName="Comms"
        selectedRoleId=""
        onIconClick={(): any => window.alert('Icon clicked')}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
