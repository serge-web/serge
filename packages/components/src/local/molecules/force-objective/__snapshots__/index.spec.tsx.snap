// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForceObjective renders correctly 1`] = `
<div
  className="force-objectives"
  style={
    Object {
      "borderColor": "#3dd0ff",
    }
  }
>
  <h3>
    Objectives
  </h3>
  <div
    className="objective-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est, praesentium sed! Animi atque incidunt maiores nisi vel voluptate? Adipisci architecto aspernatur earum error incidunt laborum magni neque odio repellendus, voluptate!
  </div>
  <div
    className="role-info"
  >
    <div
      className="role-info"
      data-tour="role-info-step"
      style={
        Object {
          "backgroundColor": "#3dd0ff",
        }
      }
    >
      <span
        className="role-type"
      >
        Comms
      </span>
      <div
        className="contain-force-skin"
        data-tour="objectives-step"
      >
        <div
          className="force-skin"
        >
          <span
            className="force-type"
          >
            Blue
          </span>
          <img
            alt=""
            className="role-icon"
            data-tour="second-step"
            onClick={[Function]}
            src="http://localhost:8080/default_img/forceDefault.png"
            style={
              Object {
                "backgroundColor": "#3dd0ff",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
